core/shared/src/main/scala/zio/prelude/AssociativeBoth.scala

```diff
@@ -65,6 +65,13 @@ object AssociativeBoth extends LawfulF.Invariant[AssociativeBothDeriveEqualInvar
       override def both[A, B](fa: => F[A], fb: => F[B]): F[(A, B)] = fa.map(a => fb.map(b => (a, b))).flatten
     }
 
+  def compose[F[+_]: AssociativeBoth: Covariant, G[_]: AssociativeBoth]
```

Comment:
I wonder if this should require that `G` by `Covariant` as well and then we can return a composed type that is both an `AssociativeBoth` and `Covariant`. They are both valid derivations but I wonder if that would be more useful. In particular that would give you a type that you could further compose if you wanted to work with even more deeply nested structures.

Reply:
I was wondering the same - I guess why not

Reply:
Although it will not be possible to do `AssociativeBoth[Chunk].compose[List].compose[Option]`.
The intermediate `Covariant` will have to be implicit.

Reply:
Done